src/frontend/README.md

```diff
@@ -1,7 +1,128 @@
-# frontend
+# frontend service
 
-Run the following command to restore dependencies to `vendor/` directory:
+The **frontend** service is responsible for rendering the UI for the store's website.
+It serves as the main entry point for the application routing requests to their
+appropriate backend services.
+The application uses Server Side Rendering (SSR) to generate HTML consumed by
+the clients, which could be web browsers, web crawlers, mobile clients or something else.
 
-```sh
-dep ensure --vendor-only
+## OpenTelemetry features
+
+| Auto-Instrumented tracing | Custom spans | Custom span attributes | Metrics | Logs |
+|---------------------------|--------------|------------------------|---------|------|
+| X                         |              | X                      |         |      |
+
+## OpenTelemetry instrumentation
+
+### Initialization
+
+The OpenTelemetry SDK is initialized in `main` using the `InitTraceProvider` function.
+
+```go
+func InitTracerProvider() *sdktrace.TracerProvider {
+    ctx := context.Background()
+
+    exporter, err := otlptracegrpc.New(ctx)
+    if err != nil {
+        log.Fatal(err)
+    }
+    tp := sdktrace.NewTracerProvider(
+        sdktrace.WithSampler(sdktrace.AlwaysSample()),
+        sdktrace.WithBatcher(exporter),
+    )
+    otel.SetTracerProvider(tp)
+    otel.SetTextMapPropagator(propagation.NewCompositeTextMapPropagator(propagation.TraceContext{}, propagation.Baggage{}))
+    return tp
+}
+```
+
+Services should call `TraceProvider.shutdown()` when the service is shutdown to
+ensure all spans are exported.
+This service makes that call as part of a deferred function in `main`.
+
+```go
+    // Initialize OpenTelemetry Tracing
+    tp := InitTracerProvider()
+    defer func() {
+        if err := tp.Shutdown(context.Background()); err != nil {
+            log.Printf("Error shutting down tracer provider: %v", err)
+        }
+    }()
+```
+
+### HTTP instrumentation
+
+This service receives HTTP requests, controlled by the gorilla/mux Router.
+The following routes are defined by the frontend:
+
+| Path              | Method | Use                               |
+|-------------------|--------|-----------------------------------|
+| `/`               | GET    | Main index page                   |
+| `/cart`           | GET    | View Cart                         |
+| `/cart`           | POST   | Add to Cart                       |
+| `/cart/checktout` | POST   | Place Order                       |
+| `/cart/empty`     | POST   | Empty Cart                        |
+| `/logout`         | GET    | Logout                            |
+| `/product/{id}`   | GET    | View Product                      |
+| `/setCurrency`    | POST   | Set Currency                      |
+| `/static/`        | *      | Static resources                  |
```

Comment:
I think the static file serving would only respond to `GET` requests

Reply:
I double-checked using curl, and am able to get the /static/favicon.ico resource with `GET` and `POST` requests.